Clients/Xamarin.Interactive.Client.Desktop/Workbook/Structure/TableOfContentsNodeExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿//
+// Author:
+//   Aaron Bockover <abock@microsoft.com>
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Xamarin.Interactive.Workbook.Structure
+{
+    static class TableOfContentsNodeExtensions
+    {
+        public static void RebuildFromJavaScript (this TableOfContentsNode node, dynamic headings)
```

Comment:
May as well remove the `this`. You can't use it as an extension method because it has a `dynamic` arg.

Reply:
Finding all the best copy pasta.